smartdashboard/src/edu/wpi/first/smartdashboard/gui/elements/MJPGStreamerViewerExtension.java

```diff
@@ -0,0 +1,215 @@
+    private static final int[] END_BYTES = new int[]{0xFF, 0xD9};
+
+    private boolean ipChanged = true;
+    private String ipString = null;
+    private double rotateAngleRad = 0;
+    private int port = 0;
+    private long lastFPSCheck = 0;
+    private int lastFPS = 0;
+    private int fpsCounter = 0;
+    public class BGThread extends Thread {
+
+        boolean destroyed = false;
+
+        public BGThread() {
+            super("Camera Viewer Background");
+        }
+
+        long lastRepaint = 0;
+        @Override
+        public void run() {
+            URLConnection connection = null;
+            InputStream stream = null;
+            ByteArrayOutputStream imageBuffer = new ByteArrayOutputStream();
+            while (!destroyed) {
+                try{
+                    System.out.println("Connecting to camera");
+                    ipChanged = false;
+                    URL url = new URL("http://"+ipString+":"+port+"/?action=stream");
+                    connection = url.openConnection();
+                    connection.setReadTimeout(250);
+                    stream = connection.getInputStream();
+
+                    while(!destroyed && !ipChanged){
+                        while(System.currentTimeMillis()-lastRepaint<10){
+                            stream.skip(stream.available());
+                            Thread.sleep(1);
+                        }
+                        stream.skip(stream.available());
+
+                        imageBuffer.reset();
+                        for(int i = 0; i<START_BYTES.length;){
+                            int b = stream.read();
+                            if(b==START_BYTES[i])
+                                i++;
+                            else
+                                i = 0;
+                        }
+                        for(int i = 0; i<START_BYTES.length;++i)
+                            imageBuffer.write(START_BYTES[i]);
+
+                        for(int i = 0; i<END_BYTES.length;){
+                            int b = stream.read();
+                            imageBuffer.write(b);
+                            if(b==END_BYTES[i])
+                                i++;
+                            else
+                                i = 0;
+                        }
+
+                        fpsCounter++;
+                        if(System.currentTimeMillis()-lastFPSCheck>500){
+                            lastFPSCheck = System.currentTimeMillis();
+                            lastFPS = fpsCounter*2;
+                            fpsCounter = 0;
+                        }
+
+                        lastRepaint = System.currentTimeMillis();
+                        ByteArrayInputStream tmpStream = new ByteArrayInputStream(imageBuffer.toByteArray());
+                        imageToDraw = ImageIO.read(tmpStream);
+                        repaint();
+                    }
+
+                } catch(Exception e){
```

Comment:
You need to handle the interrupted exception.
http://www.yegor256.com/2015/10/20/interrupted-exception.html